FRW/Services/Resources/UrlProvider.swift

```diff
@@ -0,0 +1,52 @@
+//
+//  UrlProvider.swift
+//  FRW
+//
+//  Created by Antonio Bello on 11/20/24.
+//
+
+import Foundation
+
+enum AccountType {
+    case flow, evm
+    
+    init(isEvm: Bool) {
+        self = isEvm ? .evm : .flow
+    }
+    
+    static var current: Self {
+        return EVMAccountManager.shared.selectedAccount == nil ? .flow : .evm
+    }
+}
+
+extension FlowNetworkType {
+    func getTransactionHistoryUrl(accountType: AccountType, transactionId: String) -> URL {
+        let baseUrl = getHistoryBaseUrl(accountType: accountType)
+        return URL(string: "\(baseUrl)/tx/\(transactionId)")!
+    }
+    
+    func getAccountUrl(accountType: AccountType, address: String) -> URL {
+        let baseUrl = getAccountBaseUrl(accountType: accountType)
+        return URL(string: "\(baseUrl)/account/\(address)")!
+    }
+    
+    private func getHistoryBaseUrl(accountType: AccountType) -> String {
+        return switch (accountType, self) {
+        case (.evm, .testnet): "https://evm-testnet.flowscan.io"
+        case (.evm, .mainnet): "https://evm.flowscan.io"
+        case (.evm, .previewnet): "https://previewnet.flowscan.io"
+            
+        case (.flow, .testnet): "https://testnet.flowscan.io"
+        case (.flow, .mainnet): "https://www.flowscan.io"
+        case (.flow, .previewnet): "https://previewnet.flowscan.io"
+        }
+    }
+    
+    private func getAccountBaseUrl(accountType: AccountType) -> String {
+        return switch (accountType, self) {
+        case (_, .testnet): "https://testnet.flowscan.org"
```

Comment:
EVM account doesn't share the same host with flow account. 

And host is the same as `getHistoryBaseUrl`, I think we can just use 1. 
I would recommend use enum to handle the base URL instead of functions

Reply:
Do you mean that `getAccountBaseUrl == getTransactionHistoryUrl`?

The switch enumerates all possible combinations of two different enums (`FlowNetworkType` and `AccountType`), each one resulting in a unique base url — creating a 3rd enum would still require a switch statement to pick the right one. I cant think of an elegant way to model that as a single enum with raw values.

Reply:
The Account and Transaction URL are using the same host, we don't need separate into 2 func, cause it's doing the same thing

Reply:
@lmcmz ^ done